src/edu/stanford/nlp/ie/NERFeatureFactory.java

```diff
@@ -923,237 +1021,233 @@ else if (flags.useTitle2) {
             break;
           }
         }
-        featuresC.add(featureName);
+        out.add(featureName);
       }
 
       if (flags.useABGENE) {
-        featuresC.add(c.get(CoreAnnotations.AbgeneAnnotation.class) + "-ABGENE");
-        featuresC.add(p.get(CoreAnnotations.AbgeneAnnotation.class) + "-PABGENE");
-        featuresC.add(n.get(CoreAnnotations.AbgeneAnnotation.class) + "-NABGENE");
+        out.addConcat(c.get(CoreAnnotations.AbgeneAnnotation.class), "-ABGENE");
+        out.addConcat(p.get(CoreAnnotations.AbgeneAnnotation.class), "-PABGENE");
+        out.addConcat(n.get(CoreAnnotations.AbgeneAnnotation.class), "-NABGENE");
       }
 
       if (flags.useABSTRFreqDict) {
-        featuresC.add(c.get(CoreAnnotations.AbstrAnnotation.class) + "-ABSTRACT" + c.get(CoreAnnotations.FreqAnnotation.class) + "-FREQ" + c.getString(CoreAnnotations.PartOfSpeechAnnotation.class) + "-TAG");
-        featuresC.add(c.get(CoreAnnotations.AbstrAnnotation.class) + "-ABSTRACT" + c.get(CoreAnnotations.DictAnnotation.class) + "-DICT" + c.getString(CoreAnnotations.PartOfSpeechAnnotation.class) + "-TAG");
-        featuresC.add(c.get(CoreAnnotations.AbstrAnnotation.class) + "-ABSTRACT" + c.get(CoreAnnotations.DictAnnotation.class) + "-DICT" + c.get(CoreAnnotations.FreqAnnotation.class) + "-FREQ" + c.getString(CoreAnnotations.PartOfSpeechAnnotation.class) + "-TAG");
+        out.add(c.get(CoreAnnotations.AbstrAnnotation.class) + "-ABSTRACT" + c.get(CoreAnnotations.FreqAnnotation.class) + "-FREQ" + c.getString(CoreAnnotations.PartOfSpeechAnnotation.class) + "-TAG");
```

Comment:
A little curious why no love for the 6 length features

Reply:
Largely because there are fewer of them. But it would only be consistent. I had considered a varargs call, but that again would create an object.

Reply:
Are you thinking of adding 5 & 6, or are you satisfied as is?

Reply:
Depends also on what you would prefer. Adding 5 and 6 probably means about 20 lines of code more to maintain. If these features are not used in any of the standard models, it may not be worth it.